core/modules/global/stringUtil.h

```diff
@@ -0,0 +1,69 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_STRINGUTIL_H
+#define LSST_QSERV_STRINGUTIL_H
+ /**
+  * @brief  Misc. lightweight string manipulation.
+  *
+  */
+
+namespace lsst {
+namespace qserv {
+
+/// @return true if a string is safe enough to use as a name in our SQL dialect.
+bool
+inline isNameSafe(std::string::value_type const& c) {
+    std::locale loc;
+    if(std::isalnum(c)) { // Not sure that using the default locale is safe.
+        return true;
+    }
+    switch(c) { // Special cases. '_' is the only one right now.
```

Comment:
;) we are really restrictive right now: only alnum + _, so any punctuation or quoting or escaping is sanitized out.